actionpack/lib/action_dispatch/http/mime_negotiation.rb

```diff
@@ -151,6 +154,17 @@ def valid_accept_header
       def use_accept_header
         !self.class.ignore_accept_header
       end
+
+      private
+
+      def extract_variant!
+        if parameters.key?(:format)
+          segments = parameters[:format].to_s.split("+")
```

Comment:
Currently our discussion is an comment stream hidden by default. Can you post a new comment asking if it would good to implement this on the router instead and what the implications of that would be (which is what your comment just before this one is about).